Prometheus/Exemplar.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Globalization;
+
+namespace Prometheus;
+
+public static class Exemplar
+{
+    /// <summary>
+    /// An exemplar label key.
+    /// </summary>
+    public readonly struct LabelKey
+    {
+        internal LabelKey(byte[] key, int runeCount)
+        {
+            Bytes = key;
+            RuneCount = runeCount;
+        }
+
+        internal int RuneCount { get; }
+
+        internal byte[] Bytes { get; }
+
+        /// <summary>
+        /// Create a LabelPair once a value is available
+        /// </summary>
+        public LabelPair WithValue(string value)
+        {
+            // TODO find a way of counting runes without using the StringInfo. Or find a way of pooling these :cry:
+            // It is used in K as well.
+            var si = new StringInfo(value);
+            return new LabelPair(
+                Bytes,
+                PrometheusConstants.ExportEncoding.GetBytes(value),
+                RuneCount + si.LengthInTextElements);
+        }
+    }
+
+    /// <summary>
+    /// A single exemplar label pair in a form suitable for efficient serialization.
+    /// </summary>
+    public readonly struct LabelPair
+    {
+        internal LabelPair(byte[] keyBytes, byte[] valueBytes, int runeCount)
+        {
+            KeyBytes = keyBytes;
+            ValueBytes = valueBytes;
+            RuneCount = runeCount;
+        }
+
+        internal int RuneCount { get; }
+        internal byte[] KeyBytes { get; }
+        internal byte[] ValueBytes { get; }
+    }
+    
+    /// <summary>
+    /// Return an exemplar label key, this may be curried with a value to produce a LabelPair.
+    /// </summary>
+    public static LabelKey Key(string key)
+    {
+        var si = new StringInfo(key);
```

Comment:
RFC: Need to find a better way than this.

Reply:
This approach I took above is counting graphemes (see [this](https://www.meziantou.net/how-to-correctly-count-the-number-of-characters-of-a-string.htm)), so it is incorrect.

We need to count the "Runes", the runes are unicode scalars. Runes are only available in dotnet core 3 and above.

@sandersaares could we drop support for older frameworks ?

Reply:
Hmm, what if we only allowed only ASCII characters here until we drop support for older frameworks ? I can't see any reason for an exemplar to contain anything but alphanumeric characters. We would need to assert the encoded `byte[]` and source string have same length.

Reply:
@sandersaares went with your proposed approach.